### PR TITLE
feature: add Discord and Twitch Integration

### PR DESCRIPTION
**Pull request by GarfDev**
**What**
- Add `Discord` and `Twitch` Integration

**Preview**
https://www.loom.com/share/93c9dd5209a94c88a0998d9070f90203
https://www.loom.com/share/d2dd15a51dcf4e6d98579e4f042b9f02

### CODE REVIEW

File: apps/api/src/app/config/secrets.ts [hunk 1]
```diff
@@ -91,5 +91,13 @@ export const POLYGON_RELAYER_API_SECRET = process.env.POLYGON_RELAYER_API_SECRET
 export const LOCAL_CERT = process.env.LOCAL_CERT;
 export const LOCAL_CERT_KEY = process.env.LOCAL_CERT_KEY;
 
+export const DISCORD_CLIENT_ID = process.env.DISCORD_CLIENT_ID;
```
Comment (peterpolman) [inline apps/api/src/app/config/secrets.ts]: @GarfDev Why are those secrets located in the API? I would expect them in the Auth project as redirects are handled there.

File: apps/auth/src/app/services/TwitchService.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
+import { TWITCH_CLIENT_ID, TWITCH_CLIENT_SECRET, TWITCH_REDIRECT_URI } from '@thxnetwork/api/config/secrets';
```
Comment (peterpolman) [inline apps/auth/src/app/services/TwitchService.ts]: @GarfDev Make sure to move this to auth, I see no reason for why they should be in the API project.

File: apps/auth/src/app/services/TwitchService.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        body.append('client_secret', TWITCH_CLIENT_SECRET);
+        body.append('client_id', TWITCH_CLIENT_ID);
+
+        const r = await twitchClient({
+            url: 'https://id.twitch.tv/oauth2/token',
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/x-www-form-urlencoded',
+            },
+            data: body,
```
Comment (peterpolman) [inline apps/auth/src/app/services/TwitchService.ts:67]: @GarfDev If you want to extend your URL with search query params you can just use the params property of this client which uses axios. 

Example:
```
axios({
   data: {} // will populate the body payload
   params: {} // will append the URL with query string params
})
```

File: apps/dashboard/tsconfig.app.tsbuildinfo [hunk 4]
```diff
@@ -0,0 +1 @@
...49],[72,49],[67,130],[63,131],[65,132],[64,130],[62,133]],"semanticDiagnosticsPerFile":[290,180,291,292,106,174,98,172,93,171,107,163,173,166,101,103,104,170,168,105,95,293,92,165,99,100,169,164,102,167,94,294,295,179,97,296,297,96,298,299,108,109,111,112,113,114,115,116,117,118,119,120,121,123,122,124,125,126,110,160,127,128,129,161,130,131,132,133,134,135,136,137,138,139,140,141,142,144,143,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,300,66,184,185,186,187,188,189,191,192,190,193,194,195,196,197,198,200,201,202,203,204,205,206,207,208,209,210,211,199,212,233,213,214,215,216,217,218,181,219,220,221,223,222,224,225,226,227,228,229,230,231,182,232,236,243,237,238,239,240,241,242,234,235,244,183,55,162,75,42,8,10,9,2,11,12,13,14,15,16,17,18,3,4,22,19,20,21,23,24,25,5,26,27,28,29,6,30,31,32,33,7,34,39,40,35,36,37,38,1,41,277,276,285,287,284,246,286,281,279,289,283,282,280,248,249,250,251,252,253,254,255,275,256,257,258,259,261,262,263,264,265,266,267,268,269,270,260,271,272,273,274,247,288,278,76,78,77,245,79,80,81,82,83,84,85,86,87,88,89,90,91,177,178,176,175,44,61,56,52,53,50,49,47,48,59,51,60,46,57,45,58,43,54,70,73,74,69,68,71,72,67,63,65,64,62]},"version":"4.8.4"}
```
Comment (peterpolman) [inline apps/dashboard/tsconfig.app.tsbuildinfo]: @GarfDev Those files should not be in the repo right? Please ignore and remove them.

File: apps/api/src/app/config/secrets.ts (hunk 1 above)
Comment (peterpolman) [reply apps/api/src/app/config/secrets.ts]: When you move this into the auth scope the CI will probably also be fixed.

File: apps/auth/src/app/services/TwitchService.ts (hunk 3 above)
Comment (GarfDev) [reply apps/auth/src/app/services/TwitchService.ts:67]: It is not a param, it a urlendcoded string in body to adapt with `Content-Type` type

**Comment by peterpolman** [comment]
@GarfDev Please rebase this branch so I can start a review.

**Comment by peterpolman** [comment]
@GarfDev please also add the connect buttons to the account page in auth. Perhaps you can also include the Github connect button there.

**Comment by peterpolman** [comment]
Merged into https://github.com/thxprotocol/monorepo/pull/244

**Pull request closed without merging**